src/resolvers/User.ts

```diff
@@ -19,9 +20,12 @@ import { User } from '../../node_modules/@generated/type-graphql';
 export const jwtKey =
   'superlongString_because_console_log_are_the_best_way_to_debug_this_project';
 
+const passwordRegex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/;
```

Comment:
cette portion de regex `(?=.*[a-zA-Z])` est redondante après celle-ci `(?=.*[a-z])(?=.*[A-Z])`,
tu peux l'enlever

j'aime bien la regex de Emeline car elle se comprend facilement :
` ^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})`

mais par contre la liste des caractères spéciaux est trop restreinte, il vaut mieux utiliser
`(?=.*[^a-zA-Z0-9])` (au moins un caractère qui n'est pas un chiffre ou une lettre)
```suggestion
const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9])(?=.{8,})/;
```
au moins une minuscule/ une Majuscule/ un chiffre/ un caractère spécial, et au moins 8 lettres